client/src/pages/trainer/trainees/index.tsx

```diff
@@ -5,53 +5,41 @@ import SearchBar from '@/src/shared/components/SearchBar/SearchBar';
 import Pagination from '@/src/shared/components/Pagination';
 import Table from '@/src/shared/components/Table';
 import { ListItem } from '@/src/shared/components/Table/ListItem';
+import { useGetTrainerTraineesQuery } from '@/src/services/traineeAPI';
+import { useRouter } from 'next/router';
 
 const TraineeList: React.FC = () => {
+  const router = useRouter();
   const [page, setPage] = useState(1);
+  const [search, setSearch] = useState('');
 
-  const pageSize = 10;
+  const { data } = useGetTrainerTraineesQuery({
+    searchQuery: search,
+    pageNumber: page,
+  });
 
-  const handleSearch = (): void => {
-    // Search logic here
-    'not yet implemented';
+  const handleSearch = (search: string): void => {
+    setSearch(search);
   };
 
-  const data: any = [];
-  for (let i = 1; i <= 10; i++) {
-    data.push({
-      id: i,
-      firstName: 'Jonathan',
-      lastName: 'Doeriski',
-      email: `johndoe${i}@gmail.com`,
-    });
-  }
-  // Pagination not yet implemented
-  const totalPages = Math.ceil(data.length / pageSize);
-
   const handleChangePage = (newPage: number): void => {
     setPage(newPage);
   };
 
-  const startIndex = (page - 1) * pageSize;
-  const endIndex = startIndex + pageSize;
-
-  data.slice(startIndex, endIndex);
-
   const header = [{ text: 'First Name' }, { text: 'Last Name' }, { text: 'Email' }];
 
-  const handleTableClick = (): void => {
-    // Manage Redirection to Trainee Detail here
-    console.log('clicked!');
+  const handleTableClick = (traineeId: number): void => {
+    void router.push(`/trainee/${traineeId}`);
   };
 
   const renderTraineeListItems = () => {
```

Comment:
Please consider adding `No trainees to show` and usage of `User` interface

```suggestion
  const renderTraineeListItems = () => {
    if (data && data.results.length > 0) {
      return data.results.map((trainee: User) => (
        <ListItem
          key={trainee.id}
          data={trainee}
          headerEnum={{
            'First Name': 'first_name',
            'Last Name': 'last_name',
            Email: 'email',
          }}
          showCheckbox={false}
          isAction={true}
          isClickable={true}
          handleClick={handleTableClick}
        />
      ));
    } else {
      return (
        <tr>
          <td colSpan={4} className="text-center py-4 text-gray-400">
            No trainees to show
          </td>
        </tr>
      );
    }
  };
```